main.go

```diff
@@ -84,74 +81,50 @@ func DownloadFile(downloadURL, targetPath string) error {
 	return nil
 }
 
-func main() {
-	configs := createConfigsModelFromEnvs()
-
+// downloadNugetWithVersion downloads NuGet with the given version.
+func downloadNugetWithVersion(nuGetVersion string) ([]string, error) {
```

Comment:
why do you return []string?
I expect a download function to download something and return its path.
Also, the function name and its parameters are stuttering, i would use `downloadNuGet(version string) ...`

Reply:
Renamed and refactored.